api/internal/target/kusttarget.go

```diff
@@ -304,7 +304,10 @@ func (kt *KustTarget) configureExternalGenerators() (
 				rm.Replace(r)
 			}
 		}
-		ra.AppendAll(rm)
+		err = ra.AppendAll(rm)
+		if err != nil {
+			return nil, errors.Wrapf(err, "configuring external generator '%v'", rm.Resources())
```

Comment:
Looking at this a bit more carefully, this prints out a really long and confusing error message containing the entire resource contents, which is probably unnecessary. I would suggest changing this to just 

```
return nil, errors.Wrapf(err, "configuring external generator")
```

So that the error is just `configuring external generator: may not add resource with an already registered id: ValueAnnotator.v1.transformers.example.co/notImportantHere.[noNs]`.

Same suggestion below for transformers.